internal/host/validator.go

```diff
@@ -1714,3 +1716,54 @@ func (v *validator) printVSphereUUIDEnabled(_ *validationContext, status Validat
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) serviceCanConnectToKubeAPI(c *validationContext) ValidationStatus {
+	// If we do not have a cluster then we definitely are not in Day2 status!
+	if c.cluster == nil {
+		return ValidationSuccessSuppressOutput
+	}
+
+	var status ValidationStatus
+
+	// Check for the "Day2" status "adding-hosts"
+	addingHosts := models.ClusterStatusAddingHosts
+
+	// If we are not in day2 then this validation does not apply
+	// TODO: Only applies if Kube-API is enabled - we need to add this to the check...
+	if !c.kubeApiEnabled || (c.cluster != nil && c.cluster.Status != &addingHosts) {
+		status = ValidationSuccessSuppressOutput
+	} else {
+		// The host being validated is part of a "Day2" cluster
+		// 1: Obtain the KubeAPI address
+		// TODO: Obtain this address properly - as per MGMT-10790
+		apiAddress := c.cluster.APIVipDNSName
+
+		// 2: Validate that the service can connect to kube-api - this will be needed for some Day2 operations.
+		// This will only verify host/port connectivity.
+		status = ValidationSuccess
+		apiHostPort := net.JoinHostPort(*apiAddress, "6443")
+		_, err := net.DialTimeout("tcp", apiHostPort, 10*time.Second)
+		if err != nil {
+			status = ValidationFailure
+		}
+	}
+
+	return status
+}
+
+func (v *validator) printServiceCanConnectToKubeAPI(c *validationContext, status ValidationStatus) string {
```

Comment:
Perhaps you want to merge your validations refactor PR before working on this?

Reply:
Agreed

Reply:
Just wanted to get a good feel of how this could be solved, at this stage it should really just be small changes to make it work.